src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -265,35 +264,22 @@ impl Time {
         self.time = time
     }
 
-    fn to_numeric_str(&self) -> String {
-        if self.time_type == TimeType::Date {
-            // TODO: pure calculation should be enough.
-            format!("{}", self.time.format("%Y%m%d"))
-        } else {
-            let s = self.time.format("%Y%m%d%H%M%S");
-            if self.fsp > 0 {
-                // Do we need to round the result?
-                let nanos = self.time.nanosecond() / TEN_POW[9 - self.fsp as usize];
-                format!("{}.{1:02$}", s, nanos, self.fsp as usize)
-            } else {
-                format!("{}", s)
-            }
-        }
-    }
-
+    /// Returns the `Decimal` representation of the `DateTime/Date`
+    /// TODO: remove this method after implementing `convert_datetime_to_decimal`
     pub fn to_decimal(&self) -> Result<Decimal> {
         if self.is_zero() {
             return Ok(0.into());
         }
-        let dec: Decimal = box_try!(self.to_numeric_str().parse());
-        Ok(dec)
+
+        convert_datetime_to_numeric_string(&self).parse()
```

Comment:
This calling chain looks to be wired. It should be convert function depends on data type functions, instead of data type depends on convert.

Reply:
As comment https://github.com/tikv/tikv/pull/5006#discussion_r303492580 mentioned, I will move this logic to the corresponding type as a method of it.